File: easy/1720-decode-xored-array/1720-decode-xored-array.js
/**
 * @param {number[]} encoded
 * @param {number} first
 * @return {number[]}
 */
const decode = function(A, first) {
    const res = [first];
    A.forEach((n, i) => res.push(n ^ res[i]));
    return res;
};

File: easy/1720-decode-xored-array/1720-decode-xored-array.cs
public class Solution 
{
    public int[] Decode(int[] encoded, int first) 
    {
        int[] res = new int[encoded.Length + 1];
        res[0] = first;
        for (int i = 1; i <= encoded.Length; i++)
        {
            res[i] = res[i - 1] ^ encoded[i - 1];
        }
        return res;
    }
}